crates/macro-support/src/d.rs

```diff
@@ -0,0 +1,383 @@
+pub struct ForeignItemType {
+    #[darling(multiple)]
+    pub extends: Vec<syn::Path>,
+    pub is_type_of: Option<syn::Expr>,
+    pub js_name: Option<LitStr>,
+    pub js_namespace: Option<JsNamespace>,
+    pub no_deref: Flag,
+    pub typescript_type: Option<LitStr>,
+    #[darling(multiple)]
+    pub vendor_prefix: Vec<Ident>,
+}
+
+impl_parse_attr!(syn::ForeignItemType, ForeignItemType);
+
+impl<'a> From<&'a ImplItemMethod> for ast::OperationKind {
+    fn from(opts: &'a ImplItemMethod) -> Self {
+        let mut operation_kind = ast::OperationKind::Regular;
+        if let Some(g) = &opts.getter {
+            operation_kind = ast::OperationKind::Getter(g.deref().clone().explicit());
+        }
+        if let Some(s) = &opts.setter {
+            operation_kind = ast::OperationKind::Setter(s.deref().clone().explicit());
+        }
+        if opts.indexing_getter.is_present() {
+            operation_kind = ast::OperationKind::IndexingGetter;
+        }
+        if opts.indexing_setter.is_present() {
+            operation_kind = ast::OperationKind::IndexingSetter;
+        }
+        if opts.indexing_deleter.is_present() {
+            operation_kind = ast::OperationKind::IndexingDeleter;
+        }
+
+        operation_kind
+    }
+}
+
+impl<'a> From<&'a ItemFn> for ast::OperationKind {
+    fn from(opts: &'a ItemFn) -> Self {
+        let mut operation_kind = ast::OperationKind::Regular;
+        if let Some(g) = &opts.getter {
+            operation_kind = ast::OperationKind::Getter(g.deref().clone().explicit());
+        }
+        if let Some(s) = &opts.setter {
+            operation_kind = ast::OperationKind::Setter(s.deref().clone().explicit());
+        }
+        if opts.indexing_getter.is_present() {
+            operation_kind = ast::OperationKind::IndexingGetter;
+        }
+        if opts.indexing_setter.is_present() {
+            operation_kind = ast::OperationKind::IndexingSetter;
+        }
+        if opts.indexing_deleter.is_present() {
+            operation_kind = ast::OperationKind::IndexingDeleter;
+        }
+
+        operation_kind
+    }
+}
+
+impl<'a> From<&'a ForeignItemFn> for ast::OperationKind {
+    fn from(opts: &'a ForeignItemFn) -> Self {
+        let mut operation_kind = ast::OperationKind::Regular;
+        if let Some(g) = &opts.getter {
+            operation_kind = ast::OperationKind::Getter(g.deref().clone().explicit());
+        }
+        if let Some(s) = &opts.setter {
+            operation_kind = ast::OperationKind::Setter(s.deref().clone().explicit());
+        }
+        if opts.indexing_getter.is_present() {
+            operation_kind = ast::OperationKind::IndexingGetter;
+        }
+        if opts.indexing_setter.is_present() {
+            operation_kind = ast::OperationKind::IndexingSetter;
+        }
+        if opts.indexing_deleter.is_present() {
+            operation_kind = ast::OperationKind::IndexingDeleter;
+        }
+
+        operation_kind
+    }
+}
```

Comment:
These shouldn't be copy-paste duplicated. A macro or intermediate trait would be better.